lib/msal-node/test/client/ClientCredentialClient.spec.ts

```diff
@@ -748,9 +748,10 @@ describe("ClientCredentialClient unit tests", () => {
             clientCredentialRequest
         )) as AuthenticationResult;
 
-        // Check the cache to ensure the refreshed token exists (the network request was successful).
-        // Typically, the network request may not have completed by the time the below code runs.
-        // However, the network requests are mocked in these tests, so the refreshed token should be in the cache at this point.
+        // Wait two seconds for the acquireToken and its mocked network requests to complete,
+        // then check the cache to ensure the refreshed token exists (the network request was successful)
+        await new Promise((r) => setTimeout(r, 2000));
```

Comment:
2s for a unit test is quite a lot and generally waiting in tests is a code smell. 

A better pattern to check that an internal async operation has finished is: 


1. check if some field X has been modified / condition is satisfied (I guess in your case the condition at line 755)
2. if it hasn't, wait for a short amount of time, e.g. 100ms
3. goto 1

Of course, if the whole thing takes too long, e.g. 2000ms, then bail and fail the test.